README.md

```diff
@@ -85,6 +88,6 @@ At this stage, you need to resolve any merge conflicts that may arise in your ow
 1. After your PR is merged, pull the reference branch (usually `master`) and clean up your local repo in order to keep up with the remote.
 
 
-
+<!--I think the step on line 84 can be replaced by `git pull origin master` while working in your own branch? -->
```

Comment:
I'm not sure. The [manual of git-pull](https://git-scm.com/docs/git-pull) says:

> Incorporates changes from a remote repository into the current branch

However manual also says:

> \<refspec\>
           Specifies which refs to fetch and which local refs to update. When no \<refspec\>s appear on the command line, the refs to fetch are read from remote.\<repository\>.fetch variables instead (see git-fetch(1))

My guess is that you are right, but might need a small experiment ;-)

Reply:
I did a small test.

`master` in `git pull origin master` seems just to refer to the remote 'refspec' (branch) which needs to be pulled. It is pulled in the currently checked out branch; if that is not master, a merge commit with the current branch takes place.

So, no, it cannot be replaced.

Reply:
I would think that is what you needed. 

`git pull origin master` = `git fetch origin master` + `git merge FETCH_HEAD`

`FETCH_HEAD` points to the active branch in Rstudio

`git pull origin master` fetches only the commits from the master branch on the remote repo and updates the local active branch so it is aligned with the master on the remote repo

whereas

`git pull` gets all commits from all branches on the remote repo and updates the local master with the master on the repo

It is the recommended way to do this according to the inbo-rstudio-workflow: see option 1B in https://inbo.github.io/git-course/workflow_rstudio.html

If you think the guidelines presented in https://inbo.github.io/git-course/workflow_rstudio.html will work for you/this repo, I suggest that you simply refer to this link in README.md.

Reply:
Thanks for diving deeper! I've taken a second look.

You are right indeed that the current branch is correctly updated, and that is indeed the aim for the current branch. And sure it is the easiest way to go, so that's a plus.

However, with fetch you don't update your local master, but only your pointer to the remote's master (see e.g. the output of `git branch -vva` below). Personally I rather like to have an up to date local master branch as well. I consider to make a comment on this in the RStudio workflow on that point, but if that remains unchanged, I'd rather implement your suggestion for the readme, as it does work and is more user-friendly (but as you can see, I'm not yet convinced myself; may not use it as there may be caveats).

Regarding:

> `git pull` gets all commits from all branches on the remote repo and updates the local master with the master on the repo

I'm quite sure that the second part ('updates the local master') is only true when you have checked out master and a tracking relationship exists between master and the remote master. A `git pull` with another branch checked out will do the same for that branch, when a tracking relationship exists between that branch and the remote counterpart. The first part ('gets all commits from all branches') refers to the fetch operation only, not to the merge, as far as I can tell.

Some info on the below example:

- the user's own working branch is `mybranch`, branched off from master
- branch `update` stands for another branch, which has been merged with `master` on the remote (it could be a branch made by a coworker, however in this case it was made locally as well, pushed to remote and merged with master at github)

```
$ git log --oneline --graph --decorate --all
* 91ffada (origin/update, update) Update readme
* b13e1fa (origin/master, master) Update README.md
| * 0691278 (HEAD -> mybranch, origin/mybranch) Extra commit
|/  
* f709cba Initial commit

$ git pull origin master 
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (1/1), done.
From https://github.com/florisvdh/dummy
 * branch            master     -> FETCH_HEAD
   b13e1fa..8024dc1  master     -> origin/master
Merge made by the 'recursive' strategy.
 README.md | 4 ++++
 1 file changed, 4 insertions(+)

$ git log --oneline --graph --decorate --all
*   3b3291a (HEAD -> mybranch) Merge branch 'master' of https://github.com/florisvdh/dummy into mybranch
|\  
| *   8024dc1 (origin/master) Merge pull request #1 from florisvdh/update
| |\  
| | * 91ffada (origin/update, update) Update readme
| |/  
| * b13e1fa (master) Update README.md
* | 0691278 (origin/mybranch) Extra commit
|/  
* f709cba Initial commit

$ git branch -vva
  master                  b13e1fa [origin/master: behind 2] Update README.md
* mybranch                3b3291a [origin/mybranch: ahead 4] Merge branch 'master' of https://github.com/florisvdh/dummy into mybranch
  update                  91ffada [origin/update] Update readme
  remotes/origin/master   8024dc1 Merge pull request #1 from florisvdh/update
  remotes/origin/mybranch 0691278 Extra commit
  remotes/origin/update   91ffada Update readme
```